README.md

```diff
@@ -1,212 +1,264 @@
+from pyartifactory.models.Group import Group
+
+# Create a Group
+group = Group(name="test_group", description="test_group")
+new_group = art.groups.create(group)
+
+# Update user
+group.description = "test_group_2"
+updated_group = art.groups.update(group)
+```
+
+Delete a group:
+```python
+art.groups.delete("test_group")
+```
+
+
+#### Security
+
+A set of methods for performing operations on apiKeys, passwords ...
+```python
+>>> art.security.
+art.security.create_api_key(          art.security.get_encrypted_password(  art.security.revoke_api_key(
+art.security.get_api_key(             art.security.regenerate_api_key(      art.security.revoke_user_api_key(
+```
+
+
+### Repository
+
+Get the list of repositories:
+```python
+repositories = art.repositories.list()
+```
+
+Get a single repository (Local, Virtual or Remote):
+```python
+local_repo = art.repositories.get_local_repo("local_repo_name")
+virtual_repo = art.repositories.get_virtual_repo("virtual_repo_name")
+remote_repo = art.repositories.get_remote_repo("remote_repo_name")
+```
+
+Create/Update a repository:
+```python
+from pyartifactory.models.Repository import LocalRepository, VirtualRepository, RemoteRepository
+
+# Create a repository
+local_repo = LocalRepository(key="test_local_repo")
+new_local_repo = art.repositories.create_local_repo(local_repo)
+
+# Update a repository
+local_repo.description = "test_local_repo"
+updated_local_repo = art.repositories.update_local_repo(local_repo)
+
+# Same process for Virtual and Remote repositories
+```
+
+Delete a repository:
+```python
+art.repositories.delete("test_local_repo")
+```
+
+
+#### Permission
+Get the list of permissions:
+```python
+permissions = art.permissions.list()
+```
+
+Get a single permission:
+```python
+users = art.permissions.get("test_permission")
+```
+
+Create/Update a permission:
+```python
+from pyartifactory.models.Permission import Permission
+
+# Create a permission
+permission = Permission(
+    **{
+        "name": "test_permission",
+        "repositories": ["test_repository"],
+        "principals": {
+            "users": {"test_user": ["r", "w", "n", "d"]},
+            "groups": {"developers": ["r"]},
+        },
+    }
+)
+perm = art.permissions.create(permission)
+
+# Update permission
+permission.repositories = ["test_repository_2"]
+updated_permission = art.permissions.update(permission)
+```
+
+Delete a permission:
+```python
+art.permissions.delete("test_permission")
+```
+
+
+### Artifacts 
+
+#### Deploy an artifact
+```python
+artifact = art.artifacts.deploy("<LOCAL_FILE_LOCATION>", "<ARTIFACT_PATH_IN_ARTIFACTORY>")
+# artifact = art.artifacts.deploy("Desktop/myNewFile.txt", "my-repository/my/new/artifact/directory/file.txt")
+```
+
+#### Download an artifact
+```python
+artifact = art.artifacts.download("<ARTIFACT_PATH_IN_ARTIFACTORY>")
+# artifact = art.artifacts.download("my-repository/my/new/artifact/directory/file.txt")
+# The artifact will be downloaded in the current directory
+
```

Comment:
What is returned by the `download` method? The path to the local file, the artifact properties?
Also you can remove the empty line.